src/github.com/t3chguy/riot-static/mxclient/room-state.go

```diff
@@ -1,198 +0,0 @@
-	case "m.room.canonical_alias":
-		if alias, ok := event.Content["alias"].(string); ok {
-			rs.canonicalAlias = alias
-		}
-	case "m.room.create":
-		if creator, ok := event.Content["creator"].(string); ok {
-			rs.Creator = creator
-		}
-	case "m.room.join_rules": // We do not (yet) care about m.room.join_rules
-	case "m.room.member":
-		var currentMemberState *MemberInfo
-		if currentMemberState = rs.MemberMap[stateKey]; currentMemberState == nil {
-			newMemberInfo := NewMemberInfo(stateKey)
-			currentMemberState = newMemberInfo
-			rs.MemberMap[stateKey] = newMemberInfo
-		}
-
-		if usePrevContent {
-			if membership, ok := event.PrevContent["membership"].(string); ok {
-				currentMemberState.Membership = membership
-			}
-			if avatarUrl, ok := event.PrevContent["avatar_url"].(string); ok {
-				currentMemberState.AvatarURL = *NewMXCURL(avatarUrl, rs.client.HomeserverURL.String())
-			}
-			if displayName, ok := event.PrevContent["displayname"].(string); ok {
-				currentMemberState.DisplayName = displayName
-			}
-		}
-
-		if membership, ok := event.Content["membership"].(string); ok {
-			currentMemberState.Membership = membership
-		}
-		if avatarUrl, ok := event.Content["avatar_url"].(string); ok {
-			currentMemberState.AvatarURL = *NewMXCURL(avatarUrl, rs.client.HomeserverURL.String())
-		}
-		if displayName, ok := event.Content["displayname"].(string); ok {
-			currentMemberState.DisplayName = displayName
-		}
-
-		rs.memberList = rs.CalculateMemberList()
-	case "m.room.power_levels":
-		// ez convert to powerLevels
-		if data, err := json.Marshal(event.Content); err == nil {
-			var powerLevels PowerLevels
-			err = json.Unmarshal(data, &powerLevels)
-			if err == nil {
-				rs.PowerLevels = powerLevels
-			}
-		}
-
-	case "m.room.name":
-		if name, ok := event.Content["name"].(string); ok {
-			rs.Name = name
-		}
-	case "m.room.topic":
-		if topic, ok := event.Content["topic"].(string); ok {
-			rs.Topic = topic
-		}
-	case "m.room.avatar":
-		if url, ok := event.Content["url"].(string); ok {
-			rs.AvatarURL = *NewMXCURL(url, rs.client.HomeserverURL.String())
-		}
-	}
-}
-
-func (rs *RoomState) CalculateMemberList() []*MemberInfo {
-	memberList := make([]*MemberInfo, 0, len(rs.MemberMap))
-	for _, member := range rs.MemberMap {
-		if member.Membership == "join" {
-			memberList = append(memberList, member)
-		}
-	}
-
-	return memberList
-}
-
-func (rs RoomState) Members() []*MemberInfo {
-	return rs.memberList
-}
-
-func (rs RoomState) Servers() StringIntPairList {
-	serverMap := make(map[string]int)
-	for _, member := range rs.CalculateMemberList() {
-		if mxidSplit := strings.SplitN(member.MXID, ":", 2); len(mxidSplit) == 2 {
-			serverMap[mxidSplit[1]]++
-		}
-	}
-
-	serverList := make(StringIntPairList, 0, len(serverMap))
-	for server, num := range serverMap {
-		serverList = append(serverList, StringIntPair{server, num})
-	}
-
-	sort.Sort(sort.Reverse(serverList))
```

Comment:
You know, if you had just used a specific type you could save yourself the `sort.Reverse` as you could just reverse your `Less()` impl.